doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,148 @@
+	$ ./target/release/t1_reachibility_monitor_cli
+
+If we check :code:`ConnectedNodes` we will see that this relation is empty, which make sense as we haven't populated relation :code:`Links` with anything yet:
+
+.. code-block::
+
+	>> dump ConnectedNodes;
+
+To exit DDlog shell press `[CTRL+d]`. If you are using `Windows Terminal` do not press `[CTRL+D]` (note capital `D`, `[CTRL+SHIFT+d]`) as it will open new tab.
+
+Let's fix it. We are going to insert several records into :code:`Links` relation. But first let's define a simple network topology that we are going to work through. To make our life more interesting (and just slightly more complicated) the links between nodes will not be bidirectional by default.
+
+.. image:: ./images/topology.jpg
+
+Let's start again DDlog shell and insert records to recreate the network topology above:
+
+.. code-block::
+
+	$ ./target/release/t1_reachibility_monitor_cli
+	>> start;
+	>> insert Links("Menlo Park", "Santa Barbara", true);
+	>> insert Links("Menlo Park", "Salt Lake City", true);
+	>> insert Links("Santa Barbara", "Los Angeles", true);
+	>> insert Links("Los Angeles", "Santa Barbara", true);
+	>> insert Links("Los Angeles", "Salt Lake City", true);
+	>> insert Links("Salt Lake City", "Los Angeles", true);
+	>> commit;
+	>> dump ConnectedNodes;
+	ConnectedNodes{.src = "Los Angeles", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Los Angeles", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Los Angeles", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Santa Barbara"}
+
+Something is off - some nodes are connected to themselves. But why this happened? If we take a closer look at our rule we can notice that this phenomenon actually make sense. For example, Santa Barbara is reachable from Santa Barbara through Los Angeles. While it is not necessarily horrible or wrong we may want to avoid it as it clutters the relation and the output. More notably, we definitely don't want the network traffic go to Santa Barbara from Santa Barbara through Los Angeles (in real world this actually may happen but this is completely different topic). Let's fix it by adding a filtering condition to the rule that disallows source and destination match each other. Now, the rules for calculating :code:`ConnectedNodes` look as below and see how simple it is to do that in DDlog (note that only second rule was modified, the first stay intact):

```

Comment:
before noticing that something is off you could say something about this containing the expected output.